test/end-to-end/company.js

```diff
@@ -227,6 +227,142 @@ describe("Company endpoint", () => {
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.OBJECT_ID);
+            });
+
+            test("should fail if company does not exist", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                const id = "111111111111111111111111";
+                const res = await test_agent
+                    .get(`/company/${id}/profile`)
+                    .send({ adminReason })
+                    .expect(HTTPStatus.UNPROCESSABLE_ENTITY);
+                expect(res.body).toHaveProperty("error_code", ErrorTypes.VALIDATION_ERROR);
+                expect(res.body).toHaveProperty("errors");
+                expect(res.body.errors[0]).toHaveProperty("param", "companyId");
+                expect(res.body.errors[0]).toHaveProperty("msg", ValidationReasons.COMPANY_NOT_FOUND(id));
+            });
+
+            test("should return if the offers array is empty", async () => {
+                const res = await test_agent
+                    .get(`/company/${test_company_2.id}/profile`);
+                expect(res.body).toHaveProperty("offers");
+                expect(res.body.offers).toHaveProperty("length", 0);
+            });
+
+            test("should return offers lower than X (size: 3)", async () => {
+                await test_agent
+                    .post("/auth/login")
+                    .send(test_user_admin)
+                    .expect(HTTPStatus.OK);
+
+                const offers_2 = await Promise.all(Array(3).fill(null).map((_) =>
+                    Offer.create({
+                        ...generateTestOffer({
+                            "publishDate": (new Date(Date.now() - (DAY_TO_MS))).toISOString(),
+                            "publishEndDate": (new Date(Date.now() + (DAY_TO_MS))).toISOString()
+                        }),
+                        owner: test_company_2._id,
+                        ownerName: test_company_2.name,
+                        ownerLogo: test_company_2.logo })).sort((a1, a2) => a1.publishDate > a2.publishDate));
+
+                const res = await test_agent
+                    .get(`/company/${test_company_2._id}/profile`)
+                    .expect(HTTPStatus.OK);
+                expect(res.body).toHaveProperty("offers");
+                expect(res.body.offers).toHaveProperty("length", 3);
+                offers_2.forEach((val, idx) => {
+                    expect(res.body.offers[idx].owner.toString()).toEqual(val.owner.toString());
+                    expect(res.body.offers[idx].ownerName.toString()).toEqual(val.ownerName.toString());
+                    expect(res.body.offers[idx].ownerLogo.toString()).toEqual(val.ownerLogo.toString());
+                });
+            });
+
+            test("should return if offers is X (size: 5)", async () => {
```

Comment:
```suggestion
            test("should return all offers if they match the maximum limit, async () => {
```
Don't specify the size. And use the constant defined above here as well

Reply:
You can also create a test for when there are more offers than the limit and verify if the most recent ones are returned